Sources/Clappr_iOS/Classes/Plugin/Core/Drawer/BottomDrawerPlugin.swift

```diff
@@ -38,7 +38,9 @@ open class BottomDrawerPlugin: DrawerPlugin {
         guard let core = core else { return }
         
         listenTo(core, eventName: Event.didChangeScreenOrientation.rawValue) { [weak self] _ in
-            self?.hideDrawerPlugin()
+            if let isClosed = self?.isClosed, !isClosed {
```

Comment:
I agree that it should be contained in the hide / show methods, but there's a catch. 
The way that the drawer plugin was implemented to communicate with MediaControl, would be affected by this change.

For example, if the drawer is closed and we drag it just before halfway and then let go, the drawer should snap back to the closed state.
MediaControl would be informed so it could coordinate animations and such.
And because we don't have an intermediate state, we go from the `closed` to the `closed` state, meaning that we do need to trigger the `hideDrawerPlugin` even if it's already closed.

The only path that we don't want to trigger the event again is when the orientation changes, that's why the logic is here.
But I agree that we should revisit the drawer plugins and do some refactoring to improve those and some other points.

Thanks for the head's up! 😄